torch/csrc/jit/passes/quantization.cpp

```diff
@@ -692,6 +695,140 @@ c10::optional<IValue> toTwoElementIntList(Value* v) {
+  return m;
+}
+
+std::string addChildModule(
+    script::Module module,
+    const script::Module& child_module,
+    const std::vector<std::string>& path) {
+  script::Module cur = module;
+  for (size_t i = 0; i < path.size() - 1; ++i) {
+    cur = module.attr(path[i]).toModule();
+  }
+
+  std::string cur_name = path[path.size() - 1];
+  int uid = 0;
+  std::string child_name = cur_name + "_" + c10::to_string(uid++);
+  while (module.hasattr(child_name)) {
+    child_name = cur_name + "_" + c10::to_string(uid++);
+  }
+  module.register_module(child_name, child_module.clone());
+  return child_name;
+}
+
+// Check if v is the input of the graph
+bool hitGraphInput(Value* value) {
+  bool hit_graph_input = false;
+  Graph* graph = value->owningGraph();
+  for (size_t i = 0; i < graph->inputs().size(); ++i) {
+    // calling GetAttr on an input of the graph
+    // which means the module instance we call GetAttr on is
+    // passed in as input
+    hit_graph_input |= value == graph->inputs()[i];
+  }
+  return hit_graph_input;
+}
+
+// A helper class to make uses of module unique
+class ModuleUseDeduper {
+ public:
+  ModuleUseDeduper(script::Module& module) : module_(module) {}
+  void dedup() {
+    for (auto& method : module_.get_methods()) {
+      findModuleUses(method.name());
+    }
+
+    for (auto& method : module_.get_methods()) {
+      dedupModuleUses(method.name());
+    }
+  }
+
+  // Analyze the code to record necessary information
+  // to dedup the module uses
+  void findModuleUses(const std::string& method_name) {
+    script::Method method = module_.get_method(method_name);
+    auto graph = method.graph();
+
+    std::stack<Block*> blocks_to_visit;
+    blocks_to_visit.push(graph->block());
+    Value* self = graph->inputs()[0];
+    while (!blocks_to_visit.empty()) {
+      Block* b = blocks_to_visit.top();
+      blocks_to_visit.pop();
+      for (Node* n : b->nodes()) {
+        for (Block* subblock : n->blocks()) {
+          blocks_to_visit.push(subblock);
+        }
+        if (n->kind() != prim::CallMethod) {
+          continue;
+        }
+        std::vector<std::string> path;
+        Value* instance = n->inputs()[0];
+        // Iterator to traverse back the GetAttr calls
+        Value* iter = instance;
+        // trace back the instance to recover the path of the submodule
+        while (!hitGraphInput(iter)) {
+          Node* get_attr = iter->node();
+          // record the name of GetAttr
+          path.push_back(get_attr->s(attr::name));
+          // trace back the chain of GetAttr
+          iter = get_attr->inputs()[0];
+        }
+
+        if (iter == self) {
+          value_to_path_map_[instance] = path;
+          auto m = findChildModule(module_, path);
+          if (module_set_.find(m._ivalue()) == module_set_.end()) {
+            module_set_.insert(m._ivalue());
+          } else {
+            use_to_rewrite_.push_back(instance);
+          }
+        } else {
+          GRAPH_DEBUG(
```

Comment:
Please include some additional info into the log statements. Their goal is to save you running `gdb`, but if we don't print what graph we were analyzing and what value failed this condition, we still have to run `gdb` to figure it out.